### PR TITLE
ISPN-7949 HotRod client test suite improvements

### PR DESCRIPTION
**Pull request by gustavonalle**
https://issues.jboss.org/browse/ISPN-7949

Wasn't able to reduce the time a lot, couldn't come up with a different strategy for the  *ContinuousQueryTests (where most of time is spent)

### CODE REVIEW

File: client/hotrod-client/src/test/java/org/infinispan/client/hotrod/test/HotRodClientTestingUtil.java [hunk 1]
```diff
@@ -65,30 +61,32 @@ private static boolean isBindException(Throwable e) {
       return false;
    }
 
-   public static HotRodServer startHotRodServer(EmbeddedCacheManager cacheManager, int startPort, HotRodServerConfigurationBuilder builder) {
-      // TODO: This is very rudimentary!! HotRodTestingUtil needs a more robust solution where ports are generated randomly and retries if already bound
+   private static int findFreePort() {
+      try {
+         try (ServerSocket socket = new ServerSocket(0)) {
+            return socket.getLocalPort();
```
Comment (tristantarrant) [inline client/hotrod-client/src/test/java/org/infinispan/client/hotrod/test/HotRodClientTestingUtil.java]: About time someone did this !

File: server/hotrod/src/main/java/org/infinispan/server/hotrod/HotRodServer.java [hunk 2]
```diff
@@ -187,12 +189,18 @@ public void rejectedExecution(Runnable r, ThreadPoolExecutor e) {
       }
    };
 
+   private int getWorkerThreads() {
+      Integer override = Integer.getInteger(WORKER_THREADS_SYS_PROP);
+      return override == null ? configuration.workerThreads() : override;
```
Comment (tristantarrant) [inline server/hotrod/src/main/java/org/infinispan/server/hotrod/HotRodServer.java]: You can use
`Integer.getInteger(WORKER_THREADS_SYS_PROP, configuration.workerThreads());`

Comment (gustavonalle) [inline server/hotrod/src/main/java/org/infinispan/server/hotrod/HotRodServer.java]: 👍🏻

**Comment by tristantarrant** [comment]
Well, locally it went down from 5:39 to 4:09. Not too shabby.

**Comment by tristantarrant** [comment]
Pushed to master, thanks !

**Pull request merged**